modules/contact/src/constraints/ComputeFrictionalForceLMMechanicalContact.C

```diff
@@ -29,13 +29,16 @@ ComputeFrictionalForceLMMechanicalContact::validParams()
   params.addRequiredCoupledVar("friction_lm", "The frictional Lagrange's multiplier");
   params.addCoupledVar("friction_lm_dir",
                        "The frictional Lagrange's multiplier for an addtional direction.");
-
+  params.addParam<FunctionName>(
+      "function_friction",
+      "Coupled function to evaluate friction with values from contact pressure and relative "
+      "tangential velocities");
   params.addParam<Real>("c_t", 1e0, "Numerical parameter for tangential constraints");
   params.addParam<Real>(
       "epsilon",
       1.0e-7,
       "Minimum value of contact pressure that will trigger frictional enforcement");
-  params.addRequiredParam<Real>("mu", "The friction coefficient for the Coulomb friction law");
+  params.addParam<Real>("mu", 0.5, "The friction coefficient for the Coulomb friction law");
```

Comment:
why this default choice?

Reply:
The code in the constructor and residual computation made it such a default value was never used. A default for friction values is misleading I think. So I am switching to a range checked parameter without a default value to be more consistent code-wise.